src/toolbar/sticky/stickytoolbarview.js

```diff
@@ -87,64 +141,40 @@ export default class StickyToolbarView extends ToolbarView {
 	 * @protected
 	 */
 	_checkIfShouldBeSticky() {
-		const rectElement = this.model.isSticky ?
-			this._elementPlaceholder : this.element;
-		const rect = rectElement.getBoundingClientRect();
-
-		if ( rect.top < 0 && this.model.isActive ) {
-			this._stick( rect );
-		} else {
-			this._detach();
-		}
-	}
+		const limiterRect = this._limiterRect = this.limiterElement.getBoundingClientRect();
+		const toolbarRect = this._toolbarRect = this.element.getBoundingClientRect();
 
-	/**
-	 * Sticks the toolbar to the top edge of the viewport simulating
-	 * CSS position:sticky. Also see {@link #_detach}.
-	 *
-	 * TODO: Possibly replaced by CSS in the future
-	 * http://caniuse.com/#feat=css-sticky
-	 *
-	 * @protected
-	 * @param {Object} regionRect An output of getBoundingClientRect native DOM method.
-	 */
-	_stick( regionRect ) {
-		// Setup placeholder.
-		Object.assign( this._elementPlaceholder.style, {
-			display: 'block',
-			height: regionRect.height + 'px'
-		} );
+		this.model.isSticky = limiterRect.top < 0 && this.model.isActive;
 
-		// Stick the top region.
-		Object.assign( this.element.style, {
-			// Compensate 1px border which is added when becoming "sticky".
-			width: regionRect.width + 2 + 'px',
-			marginLeft: -window.scrollX - 1 + 'px'
-		} );
+		// Stick the toolbar to the top edge of the viewport simulating CSS position:sticky.
+		// TODO: Possibly replaced by CSS in the future http://caniuse.com/#feat=css-sticky
+		if ( this.model.isSticky ) {
+			this.model.isStickyToLimiterBottom = limiterRect.bottom < toolbarRect.height;
 
-		this.model.isSticky = true;
+			if ( this.model.isStickyToLimiterBottom ) {
+				this.model.left = pixelize( limiterRect.left - document.body.getBoundingClientRect().left );
+				this.model.marginLeft = null;
+			} else {
+				this.model.left = null;
+				this.model.marginLeft = pixelize( -window.scrollX - 1 );
+			}
+		}
+		// Detach the toolbar from the top edge of the viewport.
+		else {
+			this.model.isStickyToLimiterBottom = false;
+			this.model.marginLeft = this.model.left = null;
+		}
 	}
+}
 
-	/**
-	 * Detaches the toolbar from the top edge of the viewport.
-	 * See {@link #_stick}.
-	 *
-	 * @protected
-	 */
-	_detach() {
-		// Release the placeholder.
-		Object.assign( this._elementPlaceholder.style, {
-			display: 'none'
-		} );
-
-		// Detach the top region.
-		Object.assign( this.element.style, {
-			width: 'auto',
-			marginLeft: 'auto'
-		} );
-
-		this.model.isSticky = false;
-	}
+/**
+ * Add `px` unit to the passed value.
+ *
+ * @param {String|Number} value
+ * @returns {String} Value with `px` unit,
+ */
+function pixelize( value ) {
```

Comment:
It'a duplicate of `pixelize()` from `BalloonPanelView`. Perhaps it should land in UI utils.